File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if abs(x) // 10 == 0:
            return x

        minus = ""
        if x < 0:
            minus = '-'
        
        s = list(str(abs(x)))
        s.reverse()

        while s[0] == '0':
            s = s[1:]
    
        answer = int(minus + "".join(s))

        if -2**31 < answer < 2**31 - 1:
            return answer
        else:
            return 0

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        boolean isMinus = (x < 0) ? true : false;
        int abs_x = Math.abs(x);

        String str_x = String.valueOf(abs_x);
        StringBuilder sb= new StringBuilder(str_x);
        sb.reverse();
        
        int answer;
        try{
            answer = Integer.parseInt(sb.toString());
        } catch(NumberFormatException e){
            answer = 0;
        }
        

        return (isMinus) ? answer*-1 : answer;
        
    }
}